searchlib/src/vespa/searchlib/attribute/i_document_weight_attribute.h

```diff
@@ -3,17 +3,20 @@
 #pragma once
 
 #include "postinglisttraits.h"
-
 #include <functional>
 
 namespace search {
 
-namespace query { class Node; }
-
 using DocumentWeightIterator = attribute::PostingListTraits<int32_t>::const_iterator;
 
 struct IDocumentWeightAttribute
 {
+    struct LookupKey {
+        virtual ~LookupKey() = default;
+        virtual vespalib::stringref asString() const = 0;
```

Comment:
this should probably have a warning sign about the return value being 0-terminated

Reply:
Note added.